### ISSUE TITLE
broken fragmenting already fragmented packets

### ISSUE THREAD

**Issue by bartex80** [secdev/scapy#418]
I've found another issue with fragment, but this time it is not a regression:
IP.flags and IP.frag are incorrectly computed when fragment is performed on already fragmented packet.

```python
p = fragment(Ether() / IP() / ("x"*1480*3), 1480)
q = fragment(p, 1400)
assert len(q) == 6
assert q[1][IP].flags == 1  # bug
assert q[2][IP].frag > 0 # bug
```

**Comment by bartex80** [comment]
I think the fix should looks like this:
```diff
-            if i == nb-1:
-                q[IP].flags &= ~1
-            else:
-                q[IP].flags |= 1 
-            q[IP].frag = i*fragsize/8
+            if i != nb-1:
+                q[IP].flags |= 1
+            q[IP].frag += i*fragsize/8
```

**Comment by p-l-** [comment]
Since you provide a fix, could you submit a pull request? BTW, please be PEP-8 compliant in the code you change:
```Python
            if i != nb - 1:
                q[IP].flags |= 1
            q[IP].frag += i * fragsize / 8
```

Also if you create a PR, please add your test to `test/regression.uts` just after the tests introduced in #416.

That would be perfect!

**Comment by bartex80** [comment]
OK, I will create a pull request soon.

**Comment by p-l-** [comment]
Thanks. Don't forget to add `Fixes #418` in the text for your pull request.
